File: src/main/scala/com/algorithmlx/inscribers/init/InscribersConfig.scala
package com.algorithmlx.inscribers.init

import net.minecraftforge.common.ForgeConfigSpec

object InscribersConfig {
  val BUILDER: ForgeConfigSpec.Builder = new ForgeConfigSpec.Builder

  BUILDER.push("Energy settings")

  val INSCRIBER_CAPACITY: ForgeConfigSpec.IntValue = BUILDER.defineInRange("inscriber_capacity", 1000000, 1000, Int.MaxValue)

  BUILDER.pop

  val SPEC: ForgeConfigSpec = BUILDER.build()
}

File: src/main/java/com/algorithmlx/inscribers/init/config/InscribersConfig.kt
package com.algorithmlx.inscribers.init.config

import net.minecraftforge.common.ForgeConfigSpec

object InscribersConfig {
    private val builder: ForgeConfigSpec.Builder = ForgeConfigSpec.Builder()

    init {
        builder.push("Standard Inscriber")
        builder.push("Energy Settings")
    }

    val inscriberCapacity: ForgeConfigSpec.IntValue = builder.defineInRange("inscriber_capacity", 1000000, 1000, Int.MAX_VALUE)
    val inscriberPowerInsert: ForgeConfigSpec.IntValue = builder.defineInRange("inscriber_insert", 10000, 10, Int.MAX_VALUE)

    init {
        builder.pop()
    }

    val spec: ForgeConfigSpec = builder.build()
}